data-processing-lib/python/src/data_processing/runtime/pure_python/execution_configuration.py

```diff
@@ -0,0 +1,79 @@
+# (C) Copyright IBM Corp. 2024.
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import argparse
+from typing import Any
+
+from data_processing.runtime import TransformExecutionConfiguration
+from data_processing.utils import CLIArgumentProvider, get_logger
+
+
+logger = get_logger(__name__)
+
+
+cli_prefix = "runtime_"
+
+
+class PythonTransformExecutionConfiguration(TransformExecutionConfiguration):
+    """
+    A class specifying and validating Python orchestrator configuration
+    """
+
+    def __init__(self, name: str):
+        """
+        Initialization
+        """
+        super().__init__(name=name, print_params=False)
+        self.multiprocessing = False
+        self.num_processors = 0
+
+    def add_input_params(self, parser: argparse.ArgumentParser) -> None:
+        """
+        This method adds transformer specific parameter to parser
+        :param parser: parser
+        :return:
+        """
+        parser.add_argument(f"--{cli_prefix}multiprocessing", type=bool, default=False, help="use multiprocessing pool for execution")
```

Comment:
can we make the default be multiprocessing?  Why not?  Yes, not quite backwards compatible, but not sure it would really break anything.

Reply:
and then as a default number of processors something like `available-1`?

Reply:
And/or, maybe we only need `num_processors` parameter and let the user set to 1 instead of setting multiprocessing=False.  For example, what if they set multiprocessing=True, but num_processors=1.  Yes this would be odd/broken usage, but the point is that it is ambiguous.

Reply:
I kinda like for the user to decide whether he wants to use multiprocessing explicitly to make sure there are no surprises

Reply:
becides, for small amount of files multiprocessing can be slower